### PR TITLE
AVX fix for un-aligned grid sizes

### PR DESCRIPTION
**Pull request by MissingChip**

### CODE REVIEW

**Comment by denfromufa** [comment]
@MissingChip @szaman19 there should be a test that shows a passing build after the fixes

**Comment by MissingChip** [comment]
I haven't touched this in a few days since we were talking about not messing with the API, but yes it's pretty much ready I'll just make it so it works with the actual build system and has the old API as well as the relatively minor changes I made to it.

File: src/cgridgen.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+/*
+void onto_grid(npcarray& tensor, npcarray& points, const int grid_size, const int num_channels,
+            float variance) {
+    std::vector<ssize_t> grid_shape = {(ssize_t)num_channels, grid_size, grid_size, grid_size};
+    std::vector<ssize_t> grid_strides = strides_for(grid_shape);
+    add_to_grid_c(points.shape(0), (float*)points.request().ptr, grid_shape, grid_strides, (float*)tensor.request().ptr, nullptr, variance, nullptr);
+}
+*/
+
```
Comment (szaman19) [inline src/cgridgen.cpp:67]:
```suggestion
```

**Comment by szaman19** [comment]
Also, yeah. I think I'll rebase this around PR #17 

The code still needs a bit of a clean up. Not very extendible or readable now, still. Also, naming should be better. If it's just a fix for the AVX issue, then ok with merging it in.

**Comment by MissingChip** [comment]
> Also, yeah. I think I'll rebase this around PR #17
> 
> The code still needs a bit of a clean up. Not very extendible or readable now, still. Also, naming should be better. If it's just a fix for the AVX issue, then ok with merging it in.

It started as just a fix for the AVX issue, then I also started to minimize the code. There was a fair amount of unnecessary/redundant code. Definitely still needs cleanup, naming changes etc. but yeah that can be later, especially since API isn't finalized.

**Comment by szaman19** [comment]
> > Also, yeah. I think I'll rebase this around PR #17
> > The code still needs a bit of a clean up. Not very extendible or readable now, still. Also, naming should be better. If it's just a fix for the AVX issue, then ok with merging it in.
> 
> It started as just a fix for the AVX issue, then I also started to minimize the code. There was a fair amount of unnecessary/redundant code. Definitely still needs cleanup, naming changes etc. but yeah that can be later, especially since API isn't finalized.

Sounds good. LGTM :shipit:

**Review by szaman19 — APPROVED** [review]

**Comment by MissingChip** [comment]
I also added stub functionality in the "backend" part for things like per-channel weights (not per-atom wieghts, we could also do that) and different density functions, so they would be easier to add later.

File: src/cgridgen.cpp (hunk 1 above)
Comment (MissingChip) [reply src/cgridgen.cpp:67]: haha already made that edit, haven't pushed yet

**Comment by szaman19** [comment]
I am currently leaning towards the SKlearn API approach for weighted. [Here is an example](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html#sklearn.linear_model.LinearRegression.fit)

**Pull request merged**